### DISCUSSION TITLE
Will this esp32 board work with HyperSerial?

### DISCUSSION THREAD

**Discussion by appleimperio** [awawa-dev/HyperHDR#328] [General]
Hi I want to know if this board https://smile.amazon.com/HiLetgo-Development-ESP8285-Wireless-Internet/dp/B07BK435ZW/ref=sr_1_8?crid=39E6UMZGG27HM&keywords=esp32+mini+CH340G&qid=1661719442&sprefix=esp32+mini+ch340g%2Caps%2C41&sr=8-8 will work with HyperSerial without the need of level shifter. I search in the questions for the product and it state you need a CH340G to use it. If that one doesn't work can you gat me a link of one that work.

Thank You!

**Comment by awawa-dev** [comment]
Hi
EVERY esp32/8266 module (and raspberry pi also) need a level shifter 3.3V to 5V for common LED strips like sk6812, APA102 etc. CH340G is important for the USB communication with Rpi at high speed 2Mb, not for driving the LED strip. May work without the level shifter anyway: it's a matter of luck and particular setup but you are out of the technical specification and may not work or may not be stable. Arduino uses 5V logic and doesn't need it but may be use only for small setups. 
This board from Amazon looks like mine but I can't guarantee it will work: on the picture it's cp2104 (OK) and mine uses CH9102x (even better). What you will receive it's another matter.

**Reply by appleimperio** [reply]
I already have this one with me https://www.amazon.com/dp/B081PX9YFV?psc=1&ref=ppx_yo2ov_dt_b_product_details . but what if the power of the led goes straight from the power supply like i'm not powering the led from the esp will it work without shifter. Sorry im not quite clear what a lever shifter do. Thanks

**Reply by awawa-dev** [reply]
Most LED strips **logic** works at 5V (with some tolerance depending on the power input), ESP and Raspberry Pi logic works at 3.3V (outside of the LED strip tolerance, al least if we stick to the technical documentation the lower limit is usually ~3.5V).  So LED strip without the level shifter may not understand the transmission from the ESP/Rpi completely or from time to time causing blinking.

**Reply by appleimperio** [reply]
Gotcha Thanks!
